### ISSUE TITLE
TravelerList ctor: break statements

### ISSUE THREAD

**Issue by yakra** [TravelMapping/DataProcessing#272]
When a waypoint label match is found, check whether we have 2 point_indices; if so, break, and avoid iterating thru the rest of r.point_list.

This may or may not result in noticeable time savings.

**Comment by yakra** [comment]
### lab2
* C++, 16 threads: ~2.2s -> ~2.1s, ~4.76% faster
* C++, 4 threads: ~4.4s -> ~3.8s, ~15.8% faster
* C++, 1 thread: ~15.55s -> ~13.2s, ~17.8% faster
* Python: ~25.9s -> ~20.4s, ~27.0% faster

Currently DIFFing stats, logs & traveled graphs to make sure there were no unintended side effects.

**ToDo,** test:
* a .list line with DUPLICATE_LABEL included once
* a .list line with DUPLICATE_LABEL included twice

**Comment by yakra** [comment]
DIFFs look good. WRT DUPLICATE_LABELs:

* With a given DUPLICATE_LABEL included twice (as long as the label only appears twice in the .wpt), no change:
  * We continue iterating thru `r->point_list`. The 1st wp in the file matches one label in the .list line or the other. The 2nd wp does too. We end up with 2 `point_indices`, and the traveler is credited with the distance between the 2 DUPLICATE_LABEL points.
* With a given DUPLICATE_LABEL included twice, that appears >2x in the .wpt:
  * Before: A match for every occurrence in the .wpt = >2 `point_indices`, resulting in a `Waypoint label(s) not found in line:` error.
  * After: We break upon finding the 2nd match. 2 `point_indices` means the traveler is credited with the distance between the first 2 occurrences of the DUPLICATE_LABEL.
* One DUPLICATE_LABEL + one garbage nonexistent label should behave the same way.
* With one good label + DUPLICATE_LABEL:
  * Before: 1 match for the good label + >1 matches for the duplicate = >2 `point_indices`, resulting in a `Waypoint label(s) not found in line:` error.
  * After: We break upon finding the 2nd match. 2 `point_indices` means the traveler is credited with the distance between the first 2 occurrences of either of the labels used.
  * So yeah, same thing basically.

This is all academic of course.
**Garbage In, Garbage Out.**
Ideally, DUPLICATE_LABELs should not exist in the data, or be fixed right away.

**Comment by michihdeu** [comment]
> Ideally, DUPLICATE_LABELs should not exist in the data, or be fixed right away.

We might output them as warning at the end of `datacheck.sh` so that the hwy data manager is aware of it (if he runs data check...) and might fix it directly.

I don't think that we want a update failure (when Jim runs site update)

**Comment by yakra** [comment]
That was hashed out in another issue. There are relatively few of us using datacheck.sh I believe, and an extra warning there may do little to catch the attention of people already ignoring datacheck.php.

So I'm meh on *that*, but... I'll be opening up another issue in a couple minutes ;)

**Comment by yakra** [comment]
I'd have already implemented this if not for the fact that errors due to duplicate labels become more transparent to the end user.

OTOH, a solution to to #278 may be faster & make this unnecessary. That would still require a solution to #275 though.

**Comment by yakra** [comment]
A solution to #278 is in progress, and is far faster than this, making this issue moot. Closing.
